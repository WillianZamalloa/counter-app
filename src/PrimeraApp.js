// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( {saludo, subti} ) => {

    return (
            <>
                <h1> { saludo } </h1> 
                <p>{ subti }</p>
            </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subti: 'Ninja'
}

export default PrimeraApp;
