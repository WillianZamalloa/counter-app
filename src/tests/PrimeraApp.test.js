import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import PrimeraApp from '../PrimeraApp';


describe('Pruebas en <PrimeraApp />', ()=>{
    
    test('debe de mostrar el subtitulo enviado por el props ', () => {
        const saludo = 'Hola, soy Naruto';
        const subti = 'Hola, soy un subtitulo de Naruto';
        const wrapper = shallow(
                <PrimeraApp 
                    saludo={ saludo } 
                    subti={ subti }
                />
        );

        const textoParrafo = wrapper.find('p').text();
        //console.log(textoParrafo);

        expect( textoParrafo ).toBe(subti);
    })
    
    
    
    
    
    
    // test('debe de mostrar <PrimeraApp /> correctamente ', () => {
    //     const saludo = 'Hola, soy Naruto';
    //     const subti = 'Hola, soy un subtitulo de Naruto';
    //     const wrapper = shallow(
    //             <PrimeraApp 
    //                 saludo={ saludo } 
    //                 subti={ subti }
    //             />
    //     );

    //     expect( wrapper ).toMatchSnapshot();
    // })
    
    
    
    
    // test('debe de mostrar el mensaje "Hola, soy Naruto"', () => {
    //     const saludo = 'Hola, soy Naruto';
    //     const { getByText } = render(<PrimeraApp saludo = { saludo }/>);
    //     expect(getByText(saludo)).toBeInTheDocument;
        
    // });
});