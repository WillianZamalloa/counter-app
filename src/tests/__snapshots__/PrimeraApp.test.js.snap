// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
     
    Hola, soy Naruto
     
  </h1>
  <p>
     
    Hola, soy un subtitulo de Naruto
     
  </p>
</Fragment>
`;
